pkg/proc/native/proc.go

```diff
@@ -351,3 +378,55 @@ func openRedirects(redirects [3]string, foreground bool) (stdin, stdout, stderr
 
 	return stdin, stdout, stderr, closefn, nil
 }
+
+func numValid(cctx *proc.ContinueOnceContext) int {
+	r := 0
+	for i := 0; i < cctx.NumProcs(); i++ {
+		if !cctx.Proc(i).(*nativeProcess).exited {
+			r++
+		}
+	}
+	return r
+}
+
+func procForThread(cctx *proc.ContinueOnceContext, tid int) *nativeProcess {
```

Comment:
This also feels like it should be a method on `TargetGroup` and possibly included in the interface I sketched out in another comment.

Reply:
I'm going to wait before making changes around this until the general discussion about TargetGroup / ContinueOnceContext is resolved.

Reply:
Done.